### DISCUSSION TITLE
CLI v2.0.0 Release Discussion

### DISCUSSION THREAD

**Discussion by Pradumnasaraf** [Pradumnasaraf/BioDrop-CLI#31] [Ideas]
The major parts of CLI are already done. This discussion aims to polish things up before **BREAKING CHANGES release v2.0.0**

**Comment by Pradumnasaraf** [comment]
## Minor things need to fix. If you find something feels free to add them below.

1) **If a person doesn't update the profile. The final message should be something like - "Nothing got updated" instead successful message.**

<img width="666" alt="Screenshot 2023-02-18 at 5 53 11 PM" src="https://user-images.githubusercontent.com/51878265/219865600-c46ce87b-e156-4742-bc0d-f96ed6643810.png">

2) **In the update profile, we need to add a little clear message about which testimonials it's updating, the usernames, or the person given**

<img width="429" alt="Screenshot 2023-02-18 at 5 57 21 PM" src="https://user-images.githubusercontent.com/51878265/219865739-cd7fe0ec-3e09-4959-a6a0-86d5de8fad06.png">

3) **When we delete all the elements by the update option. It should remove the empty key**

4) **There are some keys that are optional and but we ask the user to enter the value. For eg, `color` and `links` in the milestone. So either we need to ask if they wan't to add them (choice) or remove the empty keys somehow at the end before writing to the file. Hope you got what I am trying to explain.**

 <img width="486" alt="Screenshot 2023-02-20 at 7 50 06 PM" src="https://user-images.githubusercontent.com/51878265/220132697-d91e3b71-e190-414c-8053-b8401d233c2b.png">

**Comment by Pradumnasaraf** [comment]
Hey, @Ananya2001-an,

I have shared a whole new discussion because #23 has become a lot messed up. You can share your thoughts here

**Reply by Ananya2001-an** [reply]
Yeah sure

**Reply by Ananya2001-an** [reply]
I think the above mentioned problems are the only ones that are to be dealt with
1, 2, 3 will be done easily but for the 4th one I think we will have to think before doing anything. For the 4th point we can create another prompt asking the user if they want to add it or not and then take input. Accordingly we will have to make those changes in delete and update functions as well.

**Reply by Pradumnasaraf** [reply]
Great points. Thank you

**Reply by Pradumnasaraf** [reply]
I think you are missing an important message https://github.com/Pradumnasaraf/LinkFree-CLI/pull/30#issuecomment-1438353974 👀

**Comment by Pradumnasaraf** [comment]
Hey, @Ananya2001-an 

I was looking at the optional keys things. One thing I noticed is that asking users if they want to enter will add more complexity. I think we can follow something we did in the create profile one. (See the last if statement in the image). We can add if statements for optional things
 
 By this, I mean that we can ask questions every time, but in brackets, we can say (**This is optional Q .press enter to skip this**). And **if statement** will only add to JSON if something is there. 
 
 
<img width="733" alt="Screenshot 2023-02-22 at 4 09 57 PM" src="https://user-images.githubusercontent.com/51878265/220597160-903a1eb9-6874-4f9e-8e74-da3c8269ec83.png">

This will also will safeguard for future as well. What do you think?

**Reply by Pradumnasaraf** [reply]
Hope you get what I mean 😅

**Reply by Ananya2001-an** [reply]
> Hope you get what I mean 😅 

Yeah I guess I understand some part of it 😂. No I get what you mean I think that's good.

**Reply by Pradumnasaraf** [reply]
Should I explain in more detail?

**Reply by Ananya2001-an** [reply]
> Should I explain in more detail?

No I understood what you meant. Basically you are saying that like in createProfile file we had if statements for adding links, tags etc only if user has given some input for them similarly we can have all questions asked and if the input for optional keys is also given then we can keep it or else not, right?

**Reply by Pradumnasaraf** [reply]
Exactly if the user has pressed enter in that question without entering any value (blank) the value for that key in the prompt answers object will be empty and we can check with that if statement (as I did above) before adding that to jsonSchema

BTW I learn this Empty data-check with an if statement, and then push in the schema thing from you, so thank you. In v1 we had a problem with empty keys in the final JSON.

**Reply by Ananya2001-an** [reply]
> Exactly if the user has pressed enter in that question without entering any value (blank) the value for that key in the prompt answers object will be empty and we can check with that if statement (as I did above)
> 
> BTW I learn this Empty data and check with an if statement, and then push in the schema thing from you, so thank you. In v1 we had a problem with empty keys in the final JSON.

Yeah it's very handy I use it often. Also you need to stop with the thank you's already I have never been acknowledged at this level 😂

**Reply by Pradumnasaraf** [reply]
Sorry, I can't get over those etiquettes 😅 .

**Reply by Ananya2001-an** [reply]
> Sorry, I can't get over those etiquettes 😅 . 

It's ok I understand that guilt of not thanking someone for even the smallest of things. I suffer from it too.

**Reply by Pradumnasaraf** [reply]
> It's ok I understand that guilt of not thanking someone for even the smallest of things. I suffer from it too.

Exaclty. I feel, saying some kind words is huge for a mental boost for someone.

**Reply by Pradumnasaraf** [reply]
Also, should I start working on the above thing I mentioned?

**Reply by Ananya2001-an** [reply]
> Also, should I start working on the above thing I mentioned?

Yeah sure ask me for any help if you need 😊

**Reply by Pradumnasaraf** [reply]
Sure. Thank you.

BTW that looks cool. **Collaborator**
<img width="287" alt="Screenshot 2023-02-22 at 4 47 09 PM" src="https://user-images.githubusercontent.com/51878265/220605038-6de78294-f835-4262-afbd-8197fe1d8575.png">

**Reply by Ananya2001-an** [reply]
> Sure. Thank you.
> 
> BTW that looks cool. **Collaborator**
> <img width="287" alt="Screenshot 2023-02-22 at 4 47 09 PM" src="https://user-images.githubusercontent.com/51878265/220605038-6de78294-f835-4262-afbd-8197fe1d8575.png">
> 

Yeah it does thanks to u

**Reply by Pradumnasaraf** [reply]
Hey, @Ananya2001-an 

I think we are almost done with the changes. Is there anything would you like to update or enhance?

**Reply by Ananya2001-an** [reply]
Yeah for the links also we should have the group key right ?

**Reply by Pradumnasaraf** [reply]
Yes, 

It will pretty quick to implement, but I was trying to skip it, thinking, about how people will understand what it means.
I mean, I am assuming that people who will use CLI will not look at docs.
What do you think?

**Reply by Ananya2001-an** [reply]
> Yes, 
> 
> It will pretty quick to implement, but I was trying to skip it, thinking, about how people will understand what it means.
> I mean, I am assuming that people who will use CLI will not look at docs.
> What do you think?
> 

Yeah I guess leave it for now. Because for the groups I have an idea of using select option so that for each new link user doesn't have to write the group name again and again but just choose from the ones already created but I will have to think before doing it.

**Reply by Ananya2001-an** [reply]
I think we just have to fix the third minor change that we have listed earlier. That's important to do for now.

**Reply by Pradumnasaraf** [reply]
> Yeah I guess leave it for now. Because for the groups I have an idea of using select option so that for each new link user doesn't have to write the group name again and again but just choose from the ones already created but I will have to think before doing it.

Yes, You just read my mind. 

We have an even better option. Enquirer has a multiple-choice option, so we can do something like when the user created all the links, we can ask them to select all the links they want to bunch and at the end with another chain prompt give it a name

**Reply by Pradumnasaraf** [reply]
> I think we just have to fix the third minor change that we have listed earlier. That's important to do for now.

Oh yes. Would you like to fix that?

**Reply by Ananya2001-an** [reply]
> > Yeah I guess leave it for now. Because for the groups I have an idea of using select option so that for each new link user doesn't have to write the group name again and again but just choose from the ones already created but I will have to think before doing it.
> 
> 
> 
> Yes, You just read my mind. 
> 
> 
> 
> We have an even better option. Enquirer has a multiple-choice option, so we can do something like when the user created all the links, we can ask them to select all the links they want to bunch and at the end with another chain prompt give it a name 

Great we can do that later then

**Reply by Ananya2001-an** [reply]
> > I think we just have to fix the third minor change that we have listed earlier. That's important to do for now.
> 
> 
> 
> Oh yes. Would you like to fix that?

Yeah sure i will do it.

**Reply by Pradumnasaraf** [reply]
Thank you. Also, If you want to enhance anything else feel free to include that in the PR.

**Comment by Pradumnasaraf** [comment]
Hey, I find something cool. We can add this to **icons** in future

https://github.com/enquirer/enquirer#autocomplete-prompt

![autocomplete-prompt](https://user-images.githubusercontent.com/51878265/220603645-ad8208ed-8816-4142-a20f-d4e9e09f2871.gif)

**Reply by Ananya2001-an** [reply]
> Hey, I find something cool. We can add this to **icons** in future
> 
> https://github.com/enquirer/enquirer#autocomplete-prompt
> 
> ![autocomplete-prompt](https://user-images.githubusercontent.com/51878265/220603645-ad8208ed-8816-4142-a20f-d4e9e09f2871.gif)
> 

Yeah this looks cool. We can actually do a lot of customization stuff in the future to make it look a bit more organized and eye catchy I guess that will be fun

**Reply by Pradumnasaraf** [reply]
Awesome 100%

**Comment by Pradumnasaraf** [comment]
**Notes for the release**

- Halt the Releases Action - **Done**
- Prepare a draft - **Done**
- Create a PR for below
    - Remove test data - **Done**
    - Bump up the version manually in `package.json` and `package-lock.json` **Done**
    - Update CHANGELOG.md
    - Update README with the new GIF, steps, feature, and maintainer's name (Ananya)

Post Release 
- Update the dependency of linkfree-cli in LinkFree

**Comment by Pradumnasaraf** [comment]
Again congrats @Ananya2001-an 🎉

**Reply by Pradumnasaraf** [reply]
It's time to test with Linkfree now

**Reply by Ananya2001-an** [reply]
> Again congrats @Ananya2001-an 🎉

Same to you 😊

**Reply by Ananya2001-an** [reply]
> It's time to test with Linkfree now

Yessss

**Reply by Pradumnasaraf** [reply]
It's working well. I have created a PR to update the CLI in LInkFree. https://github.com/EddieHubCommunity/LinkFree/pull/5025

**Reply by Ananya2001-an** [reply]
Awesome! hope it gets merged soon.

**Reply by Pradumnasaraf** [reply]
I can merge it now, but I will not do that on my own 😅 . You can also drop a review if you want

**Reply by Ananya2001-an** [reply]
> I can merge it now, but I will not do that on my own 😅 . You can also drop a review if you want

Oh yeah I forgot you are a maintainer too 😂Sure I will review it.

**Reply by Pradumnasaraf** [reply]
You are fast, you already posted on social 🚀 😂

**Reply by Ananya2001-an** [reply]
> You are fast, you already posted on social 🚀 😂

I never wait on things like these 😂

**Reply by Pradumnasaraf** [reply]
> > You are fast, you already posted on social 🚀 😂
> 
> I never wait on things like these 😂

Haha. I will engage will your Tweet and post mine after the LinkFree PR will get merged because it will not work now and people will start jumping on the project and complaining it's not working :)

**Reply by Ananya2001-an** [reply]
Yeah I actually realized that after you raised the PR. I should have waited too. But since not many people know me that won't be a problem I guess🤣

**Reply by Pradumnasaraf** [reply]
Haha, no worries.

**Reply by Pradumnasaraf** [reply]
Raised a PR to add CLI usages steps to the LinkFree docs
https://github.com/EddieHubCommunity/LinkFree/pull/5027

**Reply by Ananya2001-an** [reply]
> Raised a PR to add CLI usages steps to the LinkFree docs
> https://github.com/EddieHubCommunity/LinkFree/pull/5027

I reviewed the changes just now. It looks great in the docs. Why have you kept it in Draft mode though?

**Reply by Pradumnasaraf** [reply]
Thank you.

Because 1st PR needs to get merged 1st. So I marked so that people can skip that for now

**Reply by Ananya2001-an** [reply]
> Thank you.
> 
> Because 1st PR needs to get merged 1st. So I marked so that people can skip that for now

Okay cool

**Reply by Pradumnasaraf** [reply]
Awesome it's merged now! https://github.com/EddieHubCommunity/LinkFree/pull/5025

**Reply by Pradumnasaraf** [reply]
Also, I think I will lock both the discussion (It will not get deleted) and just a general discussion.

**Reply by Ananya2001-an** [reply]
> Also, I think I will lock both the discussion (It will not delete) and just a general discussion.

sure

**Reply by Pradumnasaraf** [reply]
> > Also, I think I will lock both the discussion (It will not delete) and just a general discussion.
> 
> sure

Aweome, I will do that. But you are maintainers you can still comment :)

**Reply by Ananya2001-an** [reply]
> > > Also, I think I will lock both the discussion (It will not delete) and just a general discussion.
> > 
> > 
> > sure
> 
> Aweome, I will do that. But you are maintainers you can still comment :)

perks of being a maintainer ;)

**Reply by Pradumnasaraf** [reply]
> perks of being a maintainer ;)

Hehe 💯
